src/tfr_navigation/src/navigator.cpp

```diff
@@ -44,84 +48,97 @@ Navigator::Navigator(ros::NodeHandle &n,
  *      -uint8_t code corresponding to the finsal status described in 
  *      Navigation.action in the tfr_msgs package
  *      -Pose describing our final position
- *  NOTE NOT THREAD SAFE
+ *  NOTE careful with the shared pointers in this class if threading becomes
```

Comment:
Is this listed somewhere not in code in case it comes up?

Reply:
No not really, I couldn't think of a place that would be more useful to put it than the header of the class if it eventually becomes a priority.

Reply:
If you have an idea, I am all ears.

Reply:
I think it's important in a more general way for all of us to remember that _any_ form of node-to-node communication is inter-process communication and therefore can run into shared data problems.  Callback functions for Services and Actions are generally running in different processes than the data they're interacting with.

Reply:
Yeah the way I am using the const shared pointers to const objects right now is completely thread safe, but if someone get's careless in the future and starts doing something silly, they need to be aware they are in a threaded enviroment and really watch their shared memory.

Reply:
However that is a side effect of the action server in general, only if you are using the execute  callaback version of the object. If you run it using the goal callback, then it runs on the main thread of the node. See this documentation for details.http://docs.ros.org/jade/api/actionlib/html/classactionlib_1_1SimpleActionServer.html

Reply:
I'm not sure what you mean.  Can you expand on that?

Reply:
When you construct an action server you have a lot of different constructor choices, but there are two main categories:
1. _Use Execute Callback_: If you use the execute callback, it spins up automatically in a separate thread. This allows you to make blocking calls. It get's called when you send it a goal. Setting the execute callback disables the goal callback. 

2. _Use the Goal Callback_: If you use the goal callback, it does not spin up a thread, your callbacks will run on the main thread in the context of the parent node. The goal callback get's called,  when you set a goal, you can also use a preempt callback in this context. This example in the tutorials shows these two used in conjunction with a subscriber callback to take an average of some sensor data. http://wiki.ros.org/actionlib_tutorials/Tutorials/SimpleActionServer%28GoalCallbackMethod%29.